### DISCUSSION TITLE
Need help on adding fonts to headings or text

### DISCUSSION THREAD

**Discussion by kabeer10395** [react-page/react-page#1029] [Q&A] [answered]
Im not exactly sure  how to do it, But i tried doing it using **createComponentPlugin** 

```
import { pluginFactories } from "@react-page/plugins-slate";
import React from "react";
import FontSelector from "./FontSelector";

export default pluginFactories.createComponentPlugin({
  addHoverButton: true,
  addToolbarButton: true,
  type: "SetFont",
  object: "mark",
  icon: <span>Font</span>,
  label: "Set Font",
  Component: "span",
  getStyle: ({ font }) => ({ font }),
  controls: {
    type: "autoform",
    schema: {
      type: "object",
      required: ["font"],
      properties: {
        font: {
          type: "string",
          uniforms: {
            component: FontSelector,
          },
        },
      },
    },
  },
  Component: ({ children, font, attributes }) => {
    return (
      <span {...attributes} className={font}>
        {children}
      </span>
    );
  },
});
```

```
import Select from "@material-ui/core/Select";
import MenuItem from "@material-ui/core/Menu";
import { useState } from "react";

const FontSelector = () => {
  const [font, setFont] = useState("overpass");

  return (
    <Select value={font} onChange={(e) => setFont(e.target.value)}>
      <MenuItem value="overpass">overpass</MenuItem>
      <MenuItem value="lato">lato</MenuItem>
      <MenuItem value="capriola">capriola</MenuItem>
    </Select>
  );
};

export default FontSelector;
```

basically im just rendering Font selector component to get the string form the select menu and add it to **span className**
But nothing happens when i click select option , it does not open the option menu to select
Is there any other way .?

**Comment by kabeer10395** [comment, marked as answer]
ok,i didnt need that **FontSelector** component 😅😅 , just had to add **enum** property to font

```
font: {
          type: "string",
          enum: ["overpass", "lato", "capriola"],
        },
```
Is there any other way to customize font ?

**Reply by macrozone** [reply]
isn't that as simple as possible? ;-)

or is there still a problem?

by the way, your initial approach with a custom Input field did not work, because you need to wrap it in `connectField` from uniforms: https://uniforms.tools/docs/api-helpers/#connectfield

that way you can transform any component that takes a `value` and a `onChange` to an input field

**Reply by kabeer10395** [reply]
cool thx :-), ya it is way to simple.

**Reply by roker15** [reply]
can i change font size or font style of normal text? I see no options for that.
